protocol/src/main/java/net/md_5/bungee/protocol/packet/CookieResponse.java

```diff
@@ -0,0 +1,51 @@
+package net.md_5.bungee.protocol.packet;
+
+import io.netty.buffer.ByteBuf;
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.EqualsAndHashCode;
+import lombok.NoArgsConstructor;
+import net.md_5.bungee.protocol.AbstractPacketHandler;
+import net.md_5.bungee.protocol.DefinedPacket;
+import net.md_5.bungee.protocol.ProtocolConstants;
+
+@Data
+@NoArgsConstructor
+@AllArgsConstructor
+@EqualsAndHashCode(callSuper = false)
+public class CookieResponse extends DefinedPacket
+{
+
+    private String cookie;
+    private byte[] data;
+
+    @Override
+    public void read(ByteBuf buf, ProtocolConstants.Direction direction, int protocolVersion)
+    {
+        cookie = readString( buf );
+        if ( buf.readBoolean() )
```

Comment:
I think you can use the methods `readNullable` and `writeNullable` with `ByteBuf::readBoolean` etc